### PR TITLE
[Merged by Bors] - chore(Topology): tweak and factor out "pseudo-metrisable second countable spaces are Lindelöf"

### PR DESCRIPTION
**Pull request by StevenClontz**
As discussed with @grunweg, this factors out a result to its own file in order to remove a circular dependency that otherwise blocks #13176

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Compactness/PseudometrizableLindelof.lean [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+-/
+
+variable {X : Type*} [TopologicalSpace X]
+
+open Set Filter Topology TopologicalSpace
+
+instance SecondCountableTopology.ofPseudoMetrizableSpaceLindelofSpace [PseudoMetrizableSpace X]
+    [LindelofSpace X] : SecondCountableTopology X := by
+  letI : PseudoMetricSpace X := TopologicalSpace.pseudoMetrizableSpacePseudoMetric X
+  have h_dense (ε) (hpos: ε > 0): ∃ s : Set X, s.Countable ∧ ∀ x, ∃ y ∈ s, dist x y ≤ ε := by
```
Comment (grunweg) [inline Mathlib/Topology/Compactness/PseudometrizableLindelof.lean:24]: In general, using < or $\leq$ is preferred over the other direction. (mathlib3 even had a linter for this; it's an open discussion if adding this linter in mathlib4 is a good idea.) Also, please add a space around the colon --- all in all, I'd write the following:
```suggestion
  have h_dense (ε) (hpos: 0 < ε) : ∃ s : Set X, s.Countable ∧ ∀ x, ∃ y ∈ s, dist x y ≤ ε := by

File: Mathlib/Topology/Compactness/PseudometrizableLindelof.lean [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+Copyright (c) 2023 Josha Dekker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Josha Dekker
+-/
+
+import Mathlib.Topology.Metrizable.Basic
+import Mathlib.Topology.Compactness.Lindelof
+
+/-!
+# Second-countability of pseduometrizable Lindelöf spaces
```
Comment (grunweg) [inline Mathlib/Topology/Compactness/PseudometrizableLindelof.lean:11]:
```suggestion
# Second-countability of pseudometrizable Lindelöf spaces

**Review by grunweg — COMMENTED** [review]
Thanks for doing this clean-up! With my comment, I originally meant "just tweak the code in-place" (without moving it) --- because now, I have to manually review the diff of the code.

In this case, reviewing these changes was still ok: I just have two very small comments.

**Comment by StevenClontz** [comment]
> Thanks for doing this clean-up! With my comment, I originally meant "just tweak the code in-place" (without moving it) --- because now, I have to manually review the diff of the code.

Whoops, yeah, I realized that later. I did encapsulate the actual changes in 7506426 but the handy "suggest changes" feature isn't available in that view. Thanks for muddling through it anyway; hopefully this is now ready to go.

File: Mathlib/Topology/Compactness/PseudometrizableLindelof.lean [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  letI : PseudoMetricSpace X := TopologicalSpace.pseudoMetrizableSpacePseudoMetric X
+  have h_dense (ε) (hpos: 0 < ε) : ∃ s : Set X, s.Countable ∧ ∀ x, ∃ y ∈ s, dist x y ≤ ε := by
+    let U := fun (z : X) ↦ Metric.ball z ε
+    obtain ⟨t, hct, huniv⟩ := LindelofSpace.elim_nhds_subcover U
+      (fun _ ↦ (Metric.isOpen_ball).mem_nhds (Metric.mem_ball_self hpos))
+    refine ⟨t, hct, fun z ↦ ?_⟩
+    obtain ⟨y, ht, hzy⟩ : ∃ y ∈ t, z ∈ U y :=
+      exists_set_mem_of_union_eq_top t (fun i ↦ U i) huniv z
+    simp only [Metric.mem_ball, U] at hzy
+    exact ⟨y, ht, hzy.le⟩
```
Comment (faenuccio) [inline Mathlib/Topology/Compactness/PseudometrizableLindelof.lean:32]:
```suggestion
    exact ⟨y, ht, (Metric.mem_ball.mp hzy).le⟩
```

**Comment by faenuccio** [comment]
Thanks!

**Comment by faenuccio** [comment]
maintainer merge

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**